packages/atlas/src/providers/wallet/tmpwallet/signer.ts

```diff
@@ -0,0 +1,60 @@
+import { TypeRegistry } from '@polkadot/types'
+import type { Signer, SignerPayloadJSON, SignerPayloadRaw, SignerResult } from '@polkadot/types/types'
+import type { HexString } from '@polkadot/util/types'
+import SignClient from '@walletconnect/sign-client'
+import type { SessionTypes } from '@walletconnect/types'
+
+import { PolkadotNamespaceChainId } from './types'
+
+interface Signature {
+  signature: HexString
+}
+
+export class WalletConnectSigner implements Signer {
+  registry: TypeRegistry
+  client: SignClient
+  session: SessionTypes.Struct
+  chainId: PolkadotNamespaceChainId
+  id = 0
+
+  constructor(client: SignClient, session: SessionTypes.Struct, chainId: PolkadotNamespaceChainId) {
+    this.client = client
+    this.session = session
+    this.chainId = chainId
+    this.registry = new TypeRegistry()
+  }
+
+  signPayload = async (payload: SignerPayloadJSON): Promise<SignerResult> => {
+    const request = {
+      topic: this.session.topic,
+      chainId: `polkadot:${payload.genesisHash.replace('0x', '').substring(0, 32)}`,
+      request: {
+        id: 1,
+        jsonrpc: '2.0',
+        method: 'polkadot_signTransaction',
+        params: { address: payload.address, transactionPayload: payload },
+      },
+    }
+
+    const { signature } = await this.client.request<Signature>(request)
+
+    return { id: ++this.id, signature }
```

Comment:
I'm asking this for the Pioneer implementation but why does the id has to be defined manually ?

Right now in Pioneer the signer just returns the `this.client.request(...` and it seems to work (also I don't define `request.id`, I just followed [the example from the doc](https://docs.walletconnect.com/advanced/multichain/polkadot/dapp-integration-guide#polkadot-example))

Reply:
this file has been taken from the WalletConnect implementation an signer is currently left unchanged. I'd say this is used to provide a correct nonce for signed transactions.